Cargo.toml

```
@@ -84,9 +88,13 @@ macos-system-configuration = ["dep:system-configuration"]
 # Enables common types used for TLS. Useless on its own.
 __tls = ["dep:rustls-pemfile", "tokio/io-util"]
 
-# Enables common rustls code.
-# Equivalent to rustls-tls-manual-roots but shorter :)
-__rustls = ["dep:hyper-rustls", "dep:tokio-rustls", "dep:rustls", "__tls", "dep:rustls-pemfile", "rustls-pki-types"]
+# Provide common feature flags along two axes:
+# - crypto provider: ring or aws-lc
+# - root certificate provider: webpki-roots or rustls-native-certs
+__rustls_roots_webpki = ["dep:webpki-roots"]
+__rustls_roots_native = ["dep:rustls-native-certs"]
+__rustls_crypto_ring = ["rustls-base", "rustls/ring"]
+__rustls_crypto_aws_lc = ["rustls-base", "rustls/aws_lc_rs"]
```

```suggestion
__rustls_crypto_ring = ["rustls-base", "rustls?/ring"]
__rustls_crypto_aws_lc = ["rustls-base", "rustls?/aws_lc_rs"]
```

a few well-placed questionmarks seem to allow this to work on 1.63